tools/antithesis-go-generator/catalog_output.go

```diff
@@ -0,0 +1,162 @@
+		file = nil
+	}
+	if file != nil {
+		if err = file.Truncate(0); err != nil {
+			file = nil
+		}
+	}
+	return file, err
+}
+
+func assertion_name_repr(s string) string {
+	if s == "Reachable" || s == "Unreachable" {
+		return fmt.Sprintf("%s(message, values)", s)
+	}
+	return fmt.Sprintf("%s(message, cond, values)", s)
+}
+
+func hit_repr(b bool) string {
+	if !b {
+		return "not_hit"
+	}
+	return "was_hit"
+}
+
+func cond_repr(b bool) string {
+	if b {
+		return "cond_true"
+	}
+	return "cond_false"
+}
+
+func must_hit_repr(b bool) string {
+	if b {
+		return "must_be_hit"
+	}
+	return "optionally_hit"
+}
+
+func expecting_repr(b bool) string {
+	if b {
+		return "expecting_true"
+	}
+	return "expecting_false"
+}
+
+func assert_type_repr(s string) string {
+	var repr_text = "reachability_test"
+
+	switch s {
+	case "every":
+		repr_text = "universal_test"
+	case "some":
+		repr_text = "existential_test"
+	case "none":
+		repr_text = "reachability_test"
+	}
+	return repr_text
+}
+
+func generate_expects(module_name string, gen_info *GenInfo) {
+	var tmpl *template.Template
+	var err error
+
+	const expector = `package main
+
+// -----------------------------------
+// Generated by Antithesis instrumentor - do not modify
+// -----------------------------------
+
+import (
+	"{{.ExpectPackageName}}"
+)
+
+func init() {
+	const cond_false = false
+	const cond_true = !cond_false
+	const was_hit = true
+	const not_hit = !was_hit
+	const must_be_hit = true
+	const optionally_hit = !must_be_hit
+	const expecting_true = true
+	const expecting_false = !expecting_true
+	
+	const universal_test = "every"
+	const existential_test = "some"
+	const reachability_test = "none"
+
+	var no_values map[string]any = nil
+	
+	{{- range .ExpectedVals }}
+	{{- $cond := cond_repr .AssertionFuncInfo.Condition -}}
+	{{- $did_hit := hit_repr false -}}
+	{{- $must_hit := must_hit_repr .AssertionFuncInfo.MustHit -}}
+	{{- $expecting := expecting_repr .AssertionFuncInfo.Expecting -}}
+	{{- $assertion_name := assertion_name_repr .Assertion -}}
+	{{- $assert_type := assert_type_repr .AssertionFuncInfo.AssertType}}
+
+	// {{$assertion_name}}
+	assert.AssertRaw("{{.Message}}", {{$cond}}, no_values, "{{.Classname}}", "{{.Funcname}}", "{{.Filename}}", {{.Line}}, {{$did_hit}}, {{$must_hit}}, {{$expecting}}, {{$assert_type}})
+	{{- end}}
+}
+`
+
+	tmpl = template.New("expector")
+
+	tmpl = tmpl.Funcs(template.FuncMap{
+		"hit_repr":            hit_repr,
+		"cond_repr":           cond_repr,
+		"must_hit_repr":       must_hit_repr,
+		"expecting_repr":      expecting_repr,
+		"assert_type_repr":    assert_type_repr,
+		"assertion_name_repr": assertion_name_repr,
+	})
+
+	if tmpl, err = tmpl.Parse(expector); err != nil {
+		panic(err)
+	}
+
+	var out_file io.Writer
+	if out_file, err = expect_output_file(module_name); err != nil {
+		panic(err)
+	}
+
+	if err = tmpl.Execute(out_file, gen_info); err != nil {
+		panic(err)
+	}
```

Comment:
Would it make sense to split the top part of this function (lots of lines, setting up the tmpl) from the bottom part (just a couple lines, applying that tmpl to a given module name)? Just for understandability, not anything else.

Reply:
Done - reads better now.